File: Python/Easy/CheckIfEveryRowAndColumnContainsAllNumbers.py
from typing import List


class Solution:
    def checkValid(self, matrix: List[List[int]]) -> bool:
        n = len(matrix)
        seen = set()

        for row in range(n):
            seen.clear()
            for col in range(n):
                num = matrix[row][col]
                if num < 1 or num > n or num in seen:
                    return False
                seen.add(num)

        for col in range(n):
            seen.clear()
            for row in range(n):
                num = matrix[row][col]
                if num < 1 or num > n or num in seen:
                    return False
                seen.add(num)

        return True

File: JavaScript/Easy/CheckIfEveryRowAndColumnContainsAllNumbers.js
/**
 * @param {number[][]} matrix
 * @return {boolean}
 */
var checkValid = function (matrix) {
  const n = matrix.length;
  const seen = new Set();

  for (let row = 0; row < n; row++) {
    seen.clear();
    for (let col = 0; col < n; col++) {
      const num = matrix[row][col];
      if (num < 1 || num > n || seen.has(num)) return false;
      seen.add(num);
    }
  }

  for (let col = 0; col < n; col++) {
    seen.clear();
    for (let row = 0; row < n; row++) {
      const num = matrix[row][col];
      if (num < 1 || num > n || seen.has(num)) return false;
      seen.add(num);
    }
  }

  return true;
};
